### PR TITLE
feat: add esm version

### LINKED ISSUE #13181: Publish CJS + ESM
### What would you like Renovate to be able to do?

Publish both CJS + ESM so that downstream tools can migrate gradually to ESM, prior to us supporting ESM only once #9890 is done

### If you have any ideas on how this should be implemented, please tell us here.

@viceice :)

### Is this a feature you are interested in implementing yourself?

No

### PR DESCRIPTION
**Pull request by viceice**
## Changes:
- compile esm package
- fix esm compatibillity

## Context:
- closes #13181

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by viceice** [comment]
import changes required because of:

```bash
> yarn start:esm
yarn run v1.22.17
$ run-s compile:esm "node esm/renovate.js {@}"
$ node tools/build.mjs
$ node esm/renovate.js
file:///home/kriese/src/renovate/esm/renovate.js:4254
import { isAbsolute, join, parse as parse3 } from "upath";
                     ^^^^
SyntaxError: Named export 'join' not found. The requested module 'upath' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'upath';
const { isAbsolute, join, parse: parse3 } = pkg;

    at ModuleJob._instantiate (internal/modules/esm/module_job.js:124:21)
    at async ModuleJob.run (internal/modules/esm/module_job.js:179:5)
    at async Loader.import (internal/modules/esm/loader.js:178:24)
    at async Object.loadESM (internal/process/esm_loader.js:68:5)
```

File: lib/expose.cjs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// https://stackoverflow.com/a/46745166/10109857
```
Comment (viceice) [inline lib/expose.cjs:1]: I'm thinking of moving this file to repo root. 🤔

File: lib/index.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export { logger } from './logger';
```
Comment (viceice) [inline lib/index.ts:1]: @rarkins Do you need something else to be exported?

File: lib/util/git/index.ts [hunk 3]
```diff
@@ -34,6 +39,9 @@ import type {
 export { setNoVerify } from './config';
 export { setPrivateKey } from './private-key';
 
+// Todo: fix upstream types
```
Comment (viceice) [inline lib/util/git/index.ts:42]:
```suggestion
// Todo: fix upstream types https://github.com/steveukx/git-js/issues/704
```

File: lib/index.ts (hunk 2 above)
Comment (rarkins) [reply lib/index.ts:1]: I do, but maybe I won't need them long term and we can skip them for now. In the app I use util/http/github, workers/global/config/parse, util/host-rules, config, workers/repository, and util/cache

Comment (viceice) [reply lib/index.ts:1]: Ok, but deep imports are no longer possible with esm, only explicit defined exports are usable.

Comment (rarkins) [reply lib/index.ts:1]: But they can be added as-needed later, right?

Comment (viceice) [reply lib/index.ts:1]: Yes

Comment (viceice) [reply lib/index.ts:1]: I've added what's needed for our internal tools

**Comment by rarkins** [comment]
Closing due to staleness

**Pull request closed without merging**